src/community/wfs-freemarker/src/main/java/org/geoserver/wfsfreemarker/html/HtmlGetFeatureOutputFormat.java

```diff
@@ -0,0 +1,68 @@
+/* (c) 2024 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wfsfreemarker.html;
+
+import static org.geoserver.wms.featureinfo.FreeMarkerTemplateManager.OutputFormat.HTML;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+import org.geoserver.config.GeoServer;
+import org.geoserver.platform.GeoServerResourceLoader;
+import org.geoserver.platform.Operation;
+import org.geoserver.platform.ServiceException;
+import org.geoserver.wfs.WFSGetFeatureOutputFormat;
+import org.geoserver.wfs.request.FeatureCollectionResponse;
+import org.geoserver.wms.WMS;
+import org.geoserver.wms.featureinfo.FreeMarkerTemplateManager;
+import org.geoserver.wms.featureinfo.HTMLTemplateManager;
+import org.geotools.feature.FeatureCollection;
+
+/**
+ * A GetFeature response handler specialized in producing HTML data for a GetFeature request through
+ * the {@link FreeMarkerTemplateManager}.
+ */
+public class HtmlGetFeatureOutputFormat extends WFSGetFeatureOutputFormat {
```

Comment:
Since GetFeatureInfo templates can be written to support [complex features](https://docs.geoserver.org/stable/en/user/tutorials/freemarker.html#common-data-models), this class should implement ``ComplexFeatureAwareFormat``.

Reply:
Done